src/content/docs/accesstokenmanagement/workers.mdx

```diff
@@ -0,0 +1,282 @@
+    </TabItem>
+</Tabs>    
+
+### Automatic token management using HTTP Factory
+
+You can register HTTP clients with the factory that will automatically use the above client definitions to request and use access tokens.
+
+The following code registers an HTTP client called `invoices` to automatically use the `invoice.client` definition:
+
+{/* prettier-ignore */}
+<Tabs syncKey="atm-workers">
+  {/* prettier-ignore */}
+  <TabItem label="V4">
+
+    ```csharp
+    // Program.cs
+    services.AddClientCredentialsHttpClient("invoices", 
+        ClientCredentialsClientName.Parse("invoice.client"), 
+        client =>
+    {
+        client.BaseAddress = new Uri("https://apis.company.com/invoice/");
+    });
+    ```
+
+    You can also set up a typed HTTP client to use a token client definition, e.g.:
+
+    ```csharp
+    // Program.cs
+    services.AddHttpClient<CatalogClient>(client =>
+        {
+            client.BaseAddress = new Uri("https://apis.company.com/catalog/");
+        })
+        .AddClientCredentialsTokenHandler(ClientCredentialsClientName.Parse("catalog.client"));
+    ```
+  
+  </TabItem>
+  <TabItem label="V3">
+
+    ```csharp
+    // Program.cs
+    services.AddClientCredentialsHttpClient("invoices", 
+        "invoice.client", 
+        client =>
+    {
+        client.BaseAddress = new Uri("https://apis.company.com/invoice/");
+    });
+    ```
+
+    You can also set up a typed HTTP client to use a token client definition, e.g.:
+
+    ```csharp
+    // Program.cs
+    services.AddHttpClient<CatalogClient>(client =>
+        {
+            client.BaseAddress = new Uri("https://apis.company.com/catalog/");
+        })
+        .AddClientCredentialsTokenHandler("catalog.client");
+    ```
+
+    </TabItem>
+</Tabs>
+
+
+    Once you have set up HTTP clients in the HTTP factory, then no token related code is needed at all, e.g.:
+
+    ```csharp
+    // WorkerHttpClient.cs
+    public class WorkerHttpClient : BackgroundService
+    {
+        private readonly ILogger<WorkerHttpClient> _logger;
+        private readonly IHttpClientFactory _clientFactory;
+
+        public WorkerHttpClient(ILogger<WorkerHttpClient> logger, IHttpClientFactory factory)
+        {
+            _logger = logger;
+            _clientFactory = factory;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                var client = _clientFactory.CreateClient("invoices");
+                var response = await client.GetAsync("test", stoppingToken);
+
+                // rest omitted
+            }
+        }
+    }
+    ```
+
+**remark** The clients in the factory have a message handler attached to them that automatically re-tries the request in case of a `401` response code. The request get re-sent with a newly requested access token. If this still results in a `401`, the response is returned to the caller.
```

Comment:
```suggestion
:::note
The clients in the HTTP client factory have a message handler attached to them that automatically retries the request in case of a `401` response code. The request get resent with a newly requested access token. If this still results in a `401`, the response is returned to the caller.
:::
```